### PR TITLE
Python Escrow SDK

### PR DESCRIPTION
**Pull request by flopez7**
## Description

Develop the Escrow SDK

## Summary of changes

- Create EscrowClient class
- Create create_escrow method
- Create setup method
- Create create_and_setup_escrow method
- Create fund method
- Create store_results method
- Create complete method
- Create bulk_payout method
- Create cancel method
- Create abort method
- Create add_trusted_handlers method
- Create get_balance method
- Create get_manifest_url method
- Create get_results_url method
- Create get_token_address method
- Create get_status method

## How test the changes

Run make run-test inside python SDK folder

## Related issues
#443 

## Operational checklist

- [x] All new functionality is covered by tests
- [ ] Any related documentation has been changed or added

### CODE REVIEW

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py [hunk 1]
```diff
@@ -0,0 +1,516 @@
...
+        ):
+            raise EscrowClientError("Invalid address")
+        if len(recipients) != len(amounts):
+            raise EscrowClientError("Arrays must have same length")
+        if len(recipients) == 0:
+            raise EscrowClientError("Arrays must have any value")
+        if 0 in amounts:
+            raise EscrowClientError("Amounts cannot be empty")
+        if sum(amounts) > self.get_balance(escrow_address):
+            raise EscrowClientError("Escrow does not have enough balance")
```
Comment (Dzeranov) [inline packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py:309]: Let's change it to `f"Escrow does not have enough balance. Current balance: {balance}. Amounts: {sum(amounts)}"`

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+        if not URL(final_results_url):
+            raise EscrowClientError("Invalid URL")
+        if not final_results_hash:
+            raise EscrowClientError("Invalid hash")
```
Comment (Dzeranov) [inline packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py:313]: See above about a bit more informative error messages.

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+
+        Returns:
+            None
+
+        Raises:
+            EscrowClientError: If an error occurs while checking the parameters
+        """
+
+        if not Web3.isAddress(escrow_address):
+            raise EscrowClientError("Invalid address")
```
Comment (Dzeranov) [inline packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py:336]: See above about a bit more informative error messages.

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
```
Comment (Dzeranov) [inline packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py:356]: See above about a bit more informative error messages.

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+        )
+
+    def add_trusted_handlers(self, escrow_address: str, handlers: List[str]):
+        """Adds an array of addresses to the trusted handlers list.
+
+        Args:
+            escrow_address (str): Address of the escrow
+            handlers (List[str]): Array of trusted handler addresses
+        Returns:
+            str: The address of the escrow created
```
Comment (Dzeranov) [inline packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py:369]: We don't anything in this method.

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py [hunk 6]
```diff
@@ -0,0 +1,516 @@
...
+
+        Raises:
+            EscrowClientError: If an error occurs while checking the parameters
+        """
+        if not Web3.isAddress(escrow_address) or not all(
+            Web3.isAddress(handler) for handler in handlers
+        ):
+            raise EscrowClientError("Invalid address")
```
Comment (Dzeranov) [inline packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py:377]: See above about a bit more informative error messages.

**Review by Dzeranov — CHANGES_REQUESTED** [review]

**Review by Dzeranov — APPROVED** [review]

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py [hunk 7]
```diff
@@ -0,0 +1,699 @@
...
+        Returns:
+            None
+
+        Raises:
+            EscrowClientError: If an error occurs while checking the parameters
+        """
+
+        if not Web3.isAddress(escrow_address):
+            raise EscrowClientError(f"Invalid escrow address: {escrow_address}")
+        if 0 > amount:
```
Comment (leric7) [inline packages/sdk/python/human-protocol-sdk/human_protocol_sdk/escrow.py:271]:
```suggestion
        if 0 >= amount:
```

**Review by Dzeranov — APPROVED** [review]
lgtm

**Review by portuu3 — APPROVED** [review]

**Review by portuu3 — APPROVED** [review]

**Pull request merged**